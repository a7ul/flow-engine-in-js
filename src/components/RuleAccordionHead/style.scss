.rule-accordion-head{
    &.container{
        max-width : 400px;
        min-width: 200px;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: space-between;
        flex-direction: row;
        border-radius: 5px 5px 0 0;
        overflow: hidden;
        color: white;
        cursor: pointer;
        margin: 10px 0;
        padding-right: 10px;
        user-select: none;
        font-weight: bold;
    }
    &.rounded-all-corners{
      border-radius: 5px;
    }
    &.pass{
        background-color: rgb(119, 170, 0);
    }
    &.fail{
        background-color: rgb(227, 7, 20);
    }
    &.not-done{
        background-color: grey;
    }
    .rule-id{
        padding: 10px;
        &.fail{
          background-color: rgb(250, 8, 24);
        }
        &.pass{
            background-color: rgb(146, 207, 3);
        }
        &.not-done{
            background-color: lightgrey;
        }
    }
    .rule-step{
      border-radius: 50%;
      padding: 0 5px;
      color: black;
      font-size: 12px;
      background-color: white;
    }
}