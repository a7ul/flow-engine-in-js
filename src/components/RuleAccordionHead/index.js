import React from 'react';
import PropTypes from 'prop-types';
import { noop } from '../../utils/helper';
import './style.scss';

const getColorClass = (status) => {
  if (status === true) {
    return 'pass';
  } else if (status === false) {
    return 'fail';
  }
  return 'not-done';
};

const RuleAccordionHead = (props) => {
  const statusClass = getColorClass(props.ruleStatus);
  return (
    <div
      className={`rule-accordion-head container ${statusClass} ${!props.expanded ? 'rounded-all-corners' : null}`}
      onClick={() => props.toggle()}
    >
      <div className={`rule-id ${statusClass}`}>{props.ruleId}</div>
      <div className="rule-title">{props.title}</div>
      <div className="rule-step">{props.step}</div>
    </div>
  );
};
RuleAccordionHead.defaultProps = {
  ruleId: null,
  title: null,
  ruleStatus: null,
  step: null,
  toggle: noop,
  expanded: false
};
RuleAccordionHead.propTypes = {
  ruleId: PropTypes.string,
  title: PropTypes.string,
  ruleStatus: PropTypes.oneOf([true, false, null]),
  step: PropTypes.number,
  toggle: PropTypes.func,
  expanded: PropTypes.bool
};
export default RuleAccordionHead;
