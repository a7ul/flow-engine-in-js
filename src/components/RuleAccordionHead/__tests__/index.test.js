import React from 'react';
import renderer from 'react-test-renderer';
import RuleAccordionHead from '../index';

it('renders correctly', () => {
  const tree = renderer
    .create(<RuleAccordionHead />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

