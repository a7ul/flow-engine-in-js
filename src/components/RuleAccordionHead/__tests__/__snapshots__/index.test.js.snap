// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rule-accordion-head container not-done rounded-all-corners"
  onClick={[Function]}
>
  <div
    className="rule-id not-done"
  />
  <div
    className="rule-title"
  />
  <div
    className="rule-step"
  />
</div>
`;
